### PR TITLE
CLOUDSTACK-9055: fix NPE in updating Redundant State of VPC networks

### PR DESCRIPTION
**Pull request by ustcweizhou**

### CODE REVIEW

**Comment by wilderrodrigues** [comment]
@ustcweizhou 

Please, steps to reproduce/test your PR. Also, write a bit about how you got into this issue in the PR description. In addition, which integration tests did you execute?

I don't think it would that hard to elaborate a bit on a PR and give some hints to people that will end up testing it.

Cheers,
Wilder

File: server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java [hunk 1]
```diff
@@ -1015,7 +1015,7 @@ protected void updateRoutersRedundantState(final List<DomainRouterVO> routers) {
                     RedundantState state = RedundantState.UNKNOWN;
                     if (answer != null && answer.getResult()) {
                         state = answer.getState();
-                    } else {
+                    } else if (answer != null) {
                         s_logger.info("Agent response doesn't seem to be correct ==> " + answer.getResult());
                     }
```
Comment (DaanHoogland) [inline server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java]: trivial comment: looks like
```
if (answer != null) {
    if (answer.getResult()) {
        state = answer.getState();
    } else {
        s_logger.info("Agent response doesn't seem to be correct ==> " + answer.getResult());
    }
}
```

**Comment by DaanHoogland** [comment]
LGTM by code inspection. The issue is obvious and the solution as well. I do agree with @wilderrodrigues that permanent validation of the issue is needed. In this case a ref to a unit test may be enough if it exists. otherwise a marvin/integration test would be fine.

**Comment by ustcweizhou** [comment]
@wilderrodrigues @DaanHoogland This happened when one of nodes is down. I think it will happen when the cloudstack-agent is stopped or killed.

**Comment by remibergsma** [comment]
@ustcweizhou Can you please make this PR against 4.6 branch? After merge it will be fwd-merged to master. Thanks!

**Comment by ustcweizhou** [comment]
@remibergsma #1073

**Pull request closed without merging**